src/mame/konami/spy.cpp

```diff
@@ -364,6 +335,56 @@ void spy_state::spy_collision(  )
 
 		memset(m_pmcram + loopend, 0, 0x800 - loopend); // clean up for next frame
 	}
+	else
+	{
+		// Collision check program
+		
+		if (!m_pmcram[0x2])
+			return;
+		
+		count = (m_pmcram[0x0] << 8) + m_pmcram[0x1];
+			
+		for (i = 16; i < 16 + (14 * count); i += 14)
+		{
+			if (!m_pmcram[i])
+				continue;
+			
+			// Check all 3 dimensions
+			tests_failed = 3;
+			for (j = 0; j < 3 * 4; j += 4)
+			{
+				a_pos = (m_pmcram[j + 3] << 8) + m_pmcram[j + 4];	// Object A center position
+				a_size = (m_pmcram[j + 5] << 8) + m_pmcram[j + 6];	// Object A half size
+		
+				b_pos = (m_pmcram[i + j + 1] << 8) + m_pmcram[i + j + 2];	// Object B center position
+				b_size = (m_pmcram[i + j + 3] << 8) + m_pmcram[i + j + 4];	// Object B half size
```

Comment:
You can use `get_u16_be` from `"multibyte.h"` to read big Endian values, e.g. `a_pos = get_u16_be(&m_pmcram[j + 3]);` etc. here.